src/shared/cards/Card Template.js

```diff
@@ -0,0 +1,73 @@
+import Card from 'cards/Card';
+import { Set } from 'immutable';
+
+export default class CardName extends Card {
+  name = 'Card Name';
+  cost = 4;
+  VP = 3;
+  coin = 2;
+  types = new Set(['Action', 'Treasure', 'Attack', 'Reaction', 'Victory']);
+  async onPlay(player) {
+    // Vanilla Effects
+    player.draw(1);
+    player.actions++;
+    player.buys++;
+    player.money++;
+
+    // Trash card(s) in Hand, is it game.currentPlayer, or just Player?
+    const cards = await game.currentPlayer.selectCards(min, max, predicate);
+    for card in cards {
```

Comment:
```
cards.forEach(card => {
  player.trash(card);
});
```

Reply:
Is selectCards called correctly via game.currentPlayer rather than just player?

Reply:
actually `game` is not defined, you want `player.game`
and `player` is the player that played the card while `player.game.currentPlayer` is the player whose turn it is
there would only be a difference if you play a card when its not your turn somehow